pkg/cmd/cluster/create_subcmds.go

```diff
@@ -196,6 +201,50 @@ func (o *CreateSubCmdsOptions) Complete(cmd *cobra.Command) error {
 	if o.ChartInfo.ClusterDef == "" && len(o.ChartInfo.ComponentDef) == 0 {
 		return fmt.Errorf("cannot find clusterDef in cluster spec or componentDef in componentSpecs or shardingSpecs")
 	}
+
+	// Define scheduling related variables
+	var (
+		tolerations     = make([]corev1.Toleration, 0)
+		NodeLabels      = make(map[string]string)
+		podAntiAffinity = &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution:  make([]corev1.PodAffinityTerm, 0),
+			PreferredDuringSchedulingIgnoredDuringExecution: make([]corev1.WeightedPodAffinityTerm, 0),
+		}
+	)
+
+	hasSchedulingPolicy := false
+	// Build tolerations if raw toleration rules are configured
+	if o.TolerationsRaw != nil {
+		tolerationsResult, err := util.BuildTolerations(o.TolerationsRaw)
+		if err != nil {
+			return err
+		}
+		jsonData, err := json.Marshal(tolerationsResult)
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(jsonData, &tolerations)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Set node labels if specified
+	if o.NodeLabels != nil {
+		hasSchedulingPolicy = true
+		NodeLabels = o.NodeLabels
+	}
+
+	// Build pod anti-affinity if either anti-affinity rule or topology keys are specified
+	if o.PodAntiAffinity != "" || len(o.TopologyKeys) > 0 {
+		hasSchedulingPolicy = true
+		podAntiAffinity = util.BuildPodAntiAffinity(o.PodAntiAffinity, o.TopologyKeys)
+	}
+
+	// Construct the final scheduling policy combining tolerations, node labels and pod anti-affinity
+	if hasSchedulingPolicy {
+		o.schedulingPolicy = util.BuildSchedulingPolicy(tolerations, NodeLabels, podAntiAffinity)
+	}
 	return nil
```

Comment:
refer to  the `buildPodTopologySpreadConstraints4Legacy` function in kb release-09